File: java/data_structure/Queue/Queue.java
/**
 * 队列的基本操作接口
 */

public interface Queue {

	// 向队列未尾添加元素
	public void push(Object e);

	// 从队列头部取出元素
	public Object pop();

	// 返回队列头部的元素
	public Object peek();

	// 判断队列是否为空
	public boolean isEmpty();

	// 返回队列的大小
	public int size();
}

File: python/algorithm/queue/Queue.py
#!/usr/bin/env python3
# -*- coding:utf-8 -*-
# PW @ 2019-11-12 22:59:43

# 队列
class Queue:
    __container = []

    def __init__(self):
        pass

    def enqueue(self, item):
        """
        入队
        """
        self.__container.append(item)


    def dequeue(self):
        """
        出队
        """
        return self.__container.pop(0)


    def size(self) -> int:
        """
        队的长度
        """
        return len(self.__container)


    def is_empty(self) -> bool:
        """
        队列是否为空
        """
        return self.size() <= 0


if __name__ == '__main__':
    queue = Queue()
    print("is empty: ", queue.is_empty())
    queue.enqueue(10)
    queue.enqueue(20)
    queue.enqueue(30)
    print("length: ", queue.size())
    print(queue.dequeue())
    print(queue.dequeue())
    print(queue.dequeue())
    print("is empty: ", queue.is_empty())


